### PR TITLE
Clean up & document .gitignore

### PR DESCRIPTION
**Pull request by xeruf**
This reorders some entries in the gitignore and splits it into sections. There are some question marks, I'd appreciate input there - whether they are even still needed.

It also adds the long-needed ignore pattern for logs.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,68 +1,60 @@
 # Don't add any custom patterns for platform-specific files like .DS_Store
 # or generated by custom development tools like .vscode or .idea that don't
 # affect other developers. Instead add those custom patterns to your global
-# .gitignore file.
+# gitignore or to your local .git/info/exclude.
 
+# Logs, local, temp files
 *.diff
 *.log
+mixxx.log*
```
Comment (daschuer) [inline .gitignore:9]: mixxx.log is not in the source tree isn't it?

Comment (xerus2000) [reply .gitignore:9]: Whenever I execute the tests I get ten mixxxx-log files in the root directory ^^

File: .gitignore [hunk 2]
```diff
@@ -1,68 +1,60 @@
...
+lib/**/*.lib
+lib/**/*.a
 
+# ???
 *.wixobj
 *.wixpdb
 *.msi
 *.mst
 *.cab
 build/wix/subdirs/*.wxs
```
Comment (Holzhaus) [inline .gitignore:42]: All of this is obsolete IMHO. That stuff should be part of the `cmake_build` directory, not the repo root anymore.

File: .gitignore [hunk 3]
```diff
@@ -1,68 +1,60 @@
...
-!/res/keyboard/*.cfg
-
-cache
-
-lib/*/*.a
-lib/*/lib/*.a
+res/qrc_mixxx.cc
 
 mixxx
 mixxx-test
```
Comment (Holzhaus) [inline .gitignore:28]: Obsolete with cmake

File: .gitignore [hunk 4]
```diff
@@ -1,68 +1,60 @@
...
+
+# SCons
+.sconf_temp
+.sconsign.branch
+.sconsign.dblite
+cache/
 # The following 2 files are autogenerated by scons at release time
 # based on .tmpl template file for the second one.
 build/wix/bundle/bundleloc.wxs
 build/wix/ProductID.wxi
```
Comment (Holzhaus) [inline .gitignore:57]: Obsolete with Cmake

File: .gitignore (hunk 1 above)
Comment (ronso0) [reply .gitignore]: yeah, me too.
Should we rather prevent that in the first places and create logs in the build/test dir?

Comment (xerus2000) [reply .gitignore]: we should, but I think I tried that and it was a little more complex...

Comment (xerus2000) [reply .gitignore]: I think it's related to https://github.com/mixxxdj/mixxx/pull/2938

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you.

**Pull request merged**